b2sdk/sync/folder.py

```diff
@@ -230,11 +232,16 @@ def _walk_relative_paths(self, local_dir, b2_dir, reporter, policies_manager):
                 # to iterate through large folders
                 if is_file_readable(local_path, reporter):
                     file_mod_time = int(os.path.getmtime(local_path) * 1000)
+
+                    if policies_manager.should_exclude_file_version(file_mod_time):
```

Comment:
I think a policy manager may decide to exclude a file version based on something that is not `file_mod_time`. We should either rename that method so that it is obvious it filters mod times, or, preferably, make it accept a FileVersion object which will allow for more sophisticated logic, using size, path, mod time etc.

Reply:
I totally agree, but to be consequent, your comment can be also applied to `should_exclude_file` and `should_exclude_directory`. They should have different names or, preferably, accept `File` and `AbstractFolder`. In the beginning, I wanted to do it that way, but the folder object is only created for the root and we would have to rewrite the logic completely, in NBC manner. What's more, for a large directory structure, it could cause slowing down, by creating folder objects just to pass it to the `ScanPoliciesManager`.

I can change the signature of the `should_exclude_file_version`, but it won't be aligned with the rest of `ScanPoliciesManager`. Or I can change the name. You call. For me, the current API is ok, aligned with the rest.

Reply:
this was discussed separately and NBC interface has won